internal/cloud/realm/import_export_test.go

```diff
@@ -0,0 +1,165 @@
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestRealmImportExport(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	groupID := u.CloudGroupID()
+
+	t.Run("Should fail without an auth client", func(t *testing.T) {
+		client := realm.NewClient(u.RealmServerURL())
+
+		err := client.Import(groupID, primitive.NewObjectID().Hex(), realm.ImportRequest{})
+		assert.Equal(t, realm.ErrInvalidSession, err)
+	})
+
+	t.Run("With an active session", func(t *testing.T) {
+		client := newAuthClient(t)
+
+		app, appErr := client.CreateApp(groupID, "importexport-test", realm.AppMeta{})
+		assert.Nil(t, appErr)
+
+		resetPasswordURL := "http://localhost:8080/reset_password"
+		emailConfirmationURL := "http://localhost:8080/confirm_email"
+
+		t.Run("Should import an app with auth providers", func(t *testing.T) {
+			err := client.Import(groupID, app.ID, realm.ImportRequest{
+				AuthProviders: []realm.AuthProvider{
+					{Name: "api-key", Type: "api-key"},
+					{Name: "local-userpass", Type: "local-userpass", Config: map[string]interface{}{
+						"resetPasswordUrl":     resetPasswordURL,
+						"emailConfirmationUrl": emailConfirmationURL,
+					}},
+				},
+			})
+			assert.Nil(t, err)
+		})
+
+		t.Run("Should export the same app with the imported changes included", func(t *testing.T) {
+			filename, payload, err := client.Export(groupID, app.ID, realm.ExportRequest{})
+			assert.Nil(t, err)
+			defer payload.Close()
+
+			filenameMatch, matchErr := regexp.MatchString(fmt.Sprintf("%s_.*\\.zip", app.Name), filename)
+			assert.Nil(t, matchErr)
+			assert.True(t, filenameMatch, "expected exported filename to match '$appName_yyyymmddHHMMSS'")
+
+			exported := parseExportResponse(t, payload)
+
+			t.Run("And the app config contents should be as expected", func(t *testing.T) {
+				appConfig, appConfigOK := exported[realm.FileAppConfig]
+				assert.True(t, appConfigOK, "expected exported app to have file: %s", realm.FileAppConfig)
+				assert.Equal(t, fmt.Sprintf(`{
+    "app_id": %q,
+    "config_version": %s,
+    "name": "importexport-test",
+    "location": "US-VA",
+    "deployment_model": "GLOBAL",
+    "security": {},
+    "custom_user_data_config": {
+        "enabled": false
+    },
+    "sync": {
+        "development_mode_enabled": false
+    }
+}
+`, app.ClientAppID, realm.DefaultAppConfigVersion), appConfig)
+			})
+
+			t.Run("And the auth provider contents should be as expected", func(t *testing.T) {
+				apiKeyConfigFilepath := realm.FileAuthProvider("api-key")
+				apiKeyConfigPayload, apiKeyConfigOK := exported[apiKeyConfigFilepath]
+				assert.True(t, apiKeyConfigOK, "expected exported app to have file: %s", apiKeyConfigFilepath)
+
+				var apiKeyConfig map[string]interface{}
+				assert.Nil(t, json.Unmarshal([]byte(apiKeyConfigPayload), &apiKeyConfig))
+				assert.NotEqualf(t, "", apiKeyConfig["id"], "expected api-key.json to have non-blank 'id' field")
```

Comment:
does this mean we're including `id`s?

Reply:
Your suggestion around hard-coding "for source control" eliminates this now